Source/Core/arrayRemoveDuplicates.js

```diff
@@ -0,0 +1,86 @@
+/*global define*/
+define(['./defaultValue', './defined', './DeveloperError', './Math'], function(defaultValue, defined, DeveloperError, CesiumMath) {
+    'use strict';
+
+    var removeDuplicatesEpsilon = CesiumMath.EPSILON10;
+
+    /**
+     * Removes adjacent duplicate values in an array of values.
+     *
+     * @param {Object[]} values The array of values.
+     * @param {Object} objectType The object type to use when comparing values. The object type must support the equalsEpsilon(left, right, epsilon) method.
+     * @param {Boolean} [wrapAround=false] Compare the last value in the array against the first value.
+     * @returns {Object[]|undefined} A new array of values with no adjacent duplicate values or the input array if no duplicates were found.
+     *
+     * @example
+     * // Returns [(1.0, 1.0, 1.0), (2.0, 2.0, 2.0)]
+     * var values = [
+     *     new Cesium.Cartesian3(1.0, 1.0, 1.0),
+     *     new Cesium.Cartesian3(1.0, 1.0, 1.0),
+     *     new Cesium.Cartesian3(2.0, 2.0, 2.0)];
+     * var nonDuplicatevalues = Cesium.PolylinePipeline.removeDuplicates(values, Cartesian3);
+     *
+     * @example
+     * // Returns [(1.0, 1.0, 1.0), (2.0, 2.0, 2.0)]
+     * var values = [
+     *     {x:1.0, y:1.0, z:1.0},
+     *     {x:1.0, y:1.0, z:1.0},
+     *     {x:2.0, y:2.0, z:2.0}];
```

Comment:
I found after making the change, that we have tests that actually pass generic objects to polyline instead of Cartesian3. Dan also expressed concern that making this function not depend explicitly on Cartesian3.equalsEpsilon() would make it not work with generic types. That's what drove adding the type as an argument. I guess I really don't need an example showing that the generic types still work though. 
Thoughts?